window_util constexpr functions use non-constexpr cos function

The `cos` function is not `constexpr`, and thus cannot be used in `constexpr` functions.

Clang reports this error on my Mac:

```
In file included from cunumeric/nullary/window.cc:18:
In file included from ./cunumeric/nullary/window_template.inl:17:
./cunumeric/nullary/window_util.h:72:20: fatal error: constexpr function never produces a constant expression [-Winvalid-constexpr]
  constexpr double operator()(int64_t idx) const
                   ^
./cunumeric/nullary/window_util.h:76:25: note: non-constexpr function 'cos' cannot be used in a constant expression
    return 0.42 - 0.5 * cos(val) + 0.08 * cos(2 * val);
                        ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/math.h:325:15: note: declared here
extern double cos(double);
              ^
```

--- a/src/cunumeric/nullary/window_util.h
+++ b/src/cunumeric/nullary/window_util.h
@@ -69,7 +69,7 @@ struct WindowOp<WindowOpCode::BARLETT> {
 template <>
 struct WindowOp<WindowOpCode::BLACKMAN> {
   WindowOp(int64_t M, double) : alpha_(M_PI * 2 / (M - 1)) {}
-  constexpr double operator()(int64_t idx) const
+  __CUDA_HD__ double operator()(int64_t idx) const
   {
     using std::cos;
     double val = idx * alpha_;
@@ -82,7 +82,7 @@ struct WindowOp<WindowOpCode::BLACKMAN> {
 template <>
 struct WindowOp<WindowOpCode::HAMMING> {
   WindowOp(int64_t M, double) : alpha_(M_PI * 2 / (M - 1)) {}
-  constexpr double operator()(int64_t idx) const { return 0.54 - 0.46 * std::cos(idx * alpha_); }
+  __CUDA_HD__ double operator()(int64_t idx) const { return 0.54 - 0.46 * std::cos(idx * alpha_); }
 
   double alpha_;
 };
@@ -90,7 +90,7 @@ struct WindowOp<WindowOpCode::HAMMING> {
 template <>
 struct WindowOp<WindowOpCode::HANNING> {
   WindowOp(int64_t M, double) : alpha_(M_PI * 2 / (M - 1)) {}
-  constexpr double operator()(int64_t idx) const { return 0.5 - 0.5 * std::cos(idx * alpha_); }
+  __CUDA_HD__ double operator()(int64_t idx) const { return 0.5 - 0.5 * std::cos(idx * alpha_); }
 
   double alpha_;
 };

